_search-plugins/caching/request-cache.md

```diff
@@ -0,0 +1,158 @@
+---
+layout: default
+title: Index request cache
+parent: Caching
+grand_parent: Improving search performance
+nav_order: 5
+---
+
+# Index request cache
+
+The index request cache in OpenSearch is a specialized caching mechanism designed to enhance search performance by storing the results of frequently executed search queries at the shard level. This reduces the load on the cluster and improves response times for repeated searches. This cache is enabled by default and is particularly useful for read-heavy workloads where certain queries are executed frequently.
+
+The cache is automatically invalidated at the configured refresh interval. The invalidation includes document updates (including document deletions) and changes to index settings. This ensures that stale results are never returned from the cache. When the cache size exceeds its configured limit, the least recently used entries are evicted to make room for new entries.
+
+Search requests with `size=0` are cached in the request cache by default. Search requests with non-deterministic characteristics (such as `Math.random()`) or relative times (such as `now` or `new Date()`) are ineligible for caching.
+{: .note}
+
+## Configuring request caching
+
+You can configure index request cache by setting the parameters in the `opensearch.yml` configuration file or using the REST API. For more information, see [Index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/).
+
+### Settings
+
+The following table lists the index request cache settings. For more information about dynamic settings, see [Index settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index-settings/).
+
+Setting | Data type  | Default | Level | Static/Dynamic | Description
+:--- |:-----------|:--------| :--- | :--- | :---
+`indices.cache.cleanup_interval` | Time unit  | `1m` (1 minute)  | Cluster | Static | Schedules a recurring background task that cleans up expired entries from the cache at the specified interval. 
+`indices.requests.cache.size` | Percentage | `1%`      | Cluster | Static | The cache size as a percentage of the heap size (for example, to use 1% of the heap, specify `1%`). 
+`index.requests.cache.enable` | Boolean    | `true`    | Index | Dynamic | Enables or disables the request cache. 
+
+### Example
+
+To disable the request cache for an index, send the following request:
+
+```json
+PUT /my_index/_settings
+{
+  "index.requests.cache.enable": false
+}
+```
+{% include copy-curl.html %}
+
+## Caching specific requests
+
+In addition to providing index-level or cluster-level settings for request cache, you can also cache specific search requests selectively by setting the `request_cache` query parameter to `true`:
```

Comment:
At the end of the first phrase, either "the request cache" or "request caching".